services/load_adaptor_docs/README.md

```diff
@@ -0,0 +1,253 @@
+CREATE INDEX IF NOT EXISTS idx_function_name
+    ON adaptor_function_docs(function_name);
+CREATE INDEX IF NOT EXISTS idx_signature
+    ON adaptor_function_docs USING gin(to_tsvector('english', signature));
+```
+
+## Usage
+
+Provide just the adaptor name and version. The service will automatically call the `adaptor_apis` service to fetch the docs.
+
+**Basic Payload:**
+```json
+{
+  "adaptor": "kobotoolbox",
+  "version": "4.2.7"
+}
+```
+
+You can also use the full adaptor name:
+```json
+{
+  "adaptor": "@openfn/language-kobotoolbox",
+  "version": "4.2.7"
+}
+```
+
+**Skip if already exists (recommended for bulk operations):**
+```json
+{
+  "adaptor": "kobotoolbox",
+  "version": "4.2.7",
+  "skip_if_exists": true
+}
+```
+
+When `skip_if_exists` is `true`:
+- Checks database first for existing docs for this adaptor+version
+- If found, returns existing function list immediately without calling `adaptor_apis`
+- If not found, proceeds with normal fetch and upload flow
+- **Use case**: Idempotent bulk uploads, CI/CD pipelines, avoiding redundant processing
+
+**Via entry.py:**
+```bash
+python -m services.entry load_adaptor_docs -i services/load_adaptor_docs/tmp/test_upload_simple.json
+```
+
+**Via HTTP (requires running server):**
+```bash
+curl -X POST http://localhost:3000/services/load_adaptor_docs \
+  -H "Content-Type: application/json" \
+  -d '{"adaptor": "kobotoolbox", "version": "4.2.7"}'
+```
+
+**Response (new upload):**
+```json
+{
+  "success": true,
+  "adaptor": "@openfn/language-kobotoolbox",
+  "version": "4.2.7",
+  "functions_uploaded": 7,
+  "function_list": [
+    "getDeploymentInfo",
+    "getForms",
+    "getSubmissions",
+    "http.get",
+    "http.post",
+    "http.put",
+    "http.request"
+  ]
+}
+```
+
+**Response (skipped, when docs already exist):**
+```json
+{
+  "success": true,
+  "adaptor": "@openfn/language-kobotoolbox",
+  "version": "4.2.7",
+  "skipped": true,
+  "functions_uploaded": 7,
+  "function_list": [
+    "getDeploymentInfo",
+    "getForms",
+    "getSubmissions",
+    "http.get",
+    "http.post",
+    "http.put",
+    "http.request"
+  ]
+}
+```
+
+**Response (error, when docs unavailable from adaptor_apis):**
+```json
+{
+  "code": 500,
+  "type": "ADAPTOR_API_ERROR",
+  "message": "Failed to fetch docs for @openfn/language-http@3.1.11"
+}
+```
+
+## Payload Schema
+
+**Required:**
+- **`adaptor`** (string): Adaptor name - accepts either:
+  - Short form: `"kobotoolbox"`, `"dhis2"`, etc.
+  - Full form: `"@openfn/language-kobotoolbox"`, `"@openfn/language-dhis2"`, etc.
+- **`version`** (string): Adaptor version (e.g., `"4.2.7"`)
+
+**Optional:**
+- **`skip_if_exists`** (boolean): If `true`, checks database first and returns existing data if found. If `false` (default), always fetches and replaces docs.
+- **`POSTGRES_URL`** (string): PostgreSQL connection string (falls back to environment variable)
+
+### Behavior modes:
+
+**Default mode (`skip_if_exists: false` or not provided):**
```

Comment:
My instinct is that this should be `true` by default - I think it's quite rare we want to force a reload.